onnxruntime/python/tools/kernel_explorer/kernels/skip_layer_norm_test.py

```diff
@@ -0,0 +1,123 @@
+    np.random.seed(0)
+    input_x = np.random.rand(batch_size, seq_len, hidden_size).astype(dtype)
+    skip = np.random.rand(batch_size, seq_len, hidden_size).astype(dtype)
+    bias = np.random.rand(hidden_size).astype(dtype)
+    gamma = np.random.rand(hidden_size).astype(dtype)
+    beta = np.random.rand((hidden_size)).astype(dtype)
+    epsilon = 0.0005
+    output_y = np.random.rand(batch_size, seq_len, hidden_size).astype(dtype)
+
+    input_d = ke.DeviceArray(input_x)
+    skip_d = ke.DeviceArray(skip)
+    bias_d = ke.DeviceArray(bias)
+    gamma_d = ke.DeviceArray(gamma)
+    beta_d = ke.DeviceArray(beta)
+    y_d = ke.DeviceArray(output_y)
+    my_func = getattr(ke, func)
+    my_op = my_func(
+        y_d, input_d, skip_d, gamma_d, beta_d, bias_d, epsilon, hidden_size, batch_size * seq_len * hidden_size
+    )
+    if my_op.IsSupported():
+        my_op.Run()
+
+        y_d.UpdateHostNumpyArray()
+
+        y_ref = skip_layer_norm(input_x, skip, bias, gamma, beta, epsilon)
+        np.testing.assert_almost_equal(y_ref, output_y, decimal=1e-05)
+
+
+@pytest.mark.parametrize("bert_sizes", get_bert_sizes())
+def test_skip_layer_norm(bert_sizes):
+    dtypes = ["float32", "float16"]
+    for dtype in dtypes:
+        for func in dtype_to_funcs(dtype):
+            print(func)
+            run_skip_layer_norm(*bert_sizes, dtype, func)
+
+
+def profile_skip_layer_norm_func(batch_size, seq_len, hidden_size, dtype, func):
+    np.random.seed(0)
+    input_x = np.random.rand(batch_size, seq_len, hidden_size).astype(dtype)
+    skip = np.random.rand(batch_size, seq_len, hidden_size).astype(dtype)
+    gamma = np.random.rand(hidden_size).astype(dtype)
+    beta = np.random.rand(hidden_size).astype(dtype)
+    bias = np.random.rand(hidden_size).astype(dtype)
+    epsilon = 0.0005
+    output_y = np.random.rand(batch_size, seq_len, hidden_size).astype(dtype)
+
+    input_d = ke.DeviceArray(input_x)
+    skip_d = ke.DeviceArray(skip)
+    gamma_d = ke.DeviceArray(gamma)
+    beta_d = ke.DeviceArray(beta)
+    bias_d = ke.DeviceArray(bias)
+    y_d = ke.DeviceArray(output_y)
+    my_func = getattr(ke, func)
+    my_op = my_func(
+        y_d, input_d, skip_d, gamma_d, beta_d, bias_d, epsilon, hidden_size, batch_size * seq_len * hidden_size
+    )
+    if my_op.IsSupported():
+        duration = my_op.Profile()
+        print(
+            dtype,
+            batch_size,
+            seq_len,
+            hidden_size,
+            my_func,
+            f"{duration * 1000:.2f} us",
+            f"{(input_x.size * 3 + bias.size * 3) * input_x.itemsize * 1e3 / duration / 1e9:.2f} GB/s",
+        )
+
+
+def profile():
+    bert_sizes = get_bert_sizes()
+    dtypes = ["float32"]
+    for dtype in dtypes:
+        for bert_size in bert_sizes:
+            for func in dtype_to_funcs(dtype):
+                profile_skip_layer_norm_func(*bert_size, dtype, func)
+            print()
+
+
+if __name__ == "__main__":
+    profile()
```

Comment:
Prefer unittest

Reply:
@justinchuby This is not test. This file can be run directly as `pytest <this file>` for testing or `python <this_file>` for brief micro-benchmark. They are a WIP contributations toward a set of TunableOp for BERT relate OPs in ROCm EP. Currently, both ROCm EP and the TunableOp are both highly experimental. We strive for flexiblity at the moment and we shall revisit all these parts after thoes things are settled.

Reply:
SG. Thanks for explaining